test/syntax/provide_in_syntax.test.ts

```diff
@@ -19,50 +19,29 @@ describe("ProvideInSyntax", () => {
         sandbox.restore();
     });
 
-    it("Should be able to declare a binding with singleton scope", () => {
-
-        class Ninja { }
-        let inSingletonScopeExpectation = sinon.expectation.create("inSingletonScope");
-        let mockBindingInSyntax = { inSingletonScope: inSingletonScopeExpectation } as any as inversifyInterfaces.BindingInSyntax<any>;
-        let mockBind = sinon.expectation.create("bind");
-        let bindingInSyntaxFunction =
-            (bind: inversifyInterfaces.Bind, target: any) => {
-                bind<Ninja>("Ninja");
-                return mockBindingInSyntax;
-            };
-        let binding: inversifyInterfaces.Binding<any> = (<any>bindingInSyntaxFunction)._binding;
-        let provideDoneSyntax = new ProvideDoneSyntax(binding as any);
-
-        let provideInSyntax = new ProvideInSyntax(bindingInSyntaxFunction, provideDoneSyntax);
-
-        provideInSyntax.inSingletonScope().done()(Ninja);
-        let metadata = Reflect.getMetadata(METADATA_KEY.provide, Reflect)[0];
-        metadata.constraint(mockBind);
-        expect(inSingletonScopeExpectation.calledOnce).to.eql(true, "inSingletonScope was not called exactly once");
-        expect(mockBind.calledWith("Ninja")).to.be.eql(true, "mock bind was not called");
-
-    });
-    it("Should be able to declare a binding with transient scope", () => {
-
-        class Ninja { }
-        let inTransientScopeExpectation = sinon.expectation.create("inTransientScope");
-        let mockBindingInSyntax = { inTransientScope: inTransientScopeExpectation } as any as inversifyInterfaces.BindingInSyntax<any>;
-        let mockBind = sinon.expectation.create("bind");
-        let bindingInSyntaxFunction =
-            (bind: inversifyInterfaces.Bind, target: any) => {
-                bind<Ninja>("Ninja");
-                return mockBindingInSyntax;
-            };
-        let binding: inversifyInterfaces.Binding<any> = (<any>bindingInSyntaxFunction)._binding;
-        let provideDoneSyntax = new ProvideDoneSyntax(binding as any);
-
-        let provideInSyntax = new ProvideInSyntax(bindingInSyntaxFunction, provideDoneSyntax);
-
-        provideInSyntax.inTransientScope().done()(Ninja);
-        let metadata = Reflect.getMetadata(METADATA_KEY.provide, Reflect)[0];
-        metadata.constraint(mockBind);
-        expect(inTransientScopeExpectation.calledOnce).to.eql(true, "inTransientScope was not called exactly once");
-        expect(mockBind.calledWith("Ninja")).to.be.eql(true, "mock bind was not called");
-
+    ["inSingletonScope", "inRequestScope", "inTransientScope"].forEach(scope => {
```

Comment:
I combined the shared aspects of the code into one repeated test (the equivalent of a data provider). 

The weirdest part was calling `provideInSyntax.in[...]Scope()`. I haven't called a string as a function in typescript so I ended up defining it as an object with string keys to get around the `no-implicit-any` compiler issue. I'm open to other approaches.